driver/restdriver.go

```diff
@@ -37,19 +52,422 @@ func (driver *RestDriver) Initialize(logger logger.LoggingClient, asyncValues ch
+			return nil, fmt.Errorf(castError, resource.Name, err)
+		}
+		if err := checkUintValueRange(valueType, val); err != nil {
+			return nil, err
+		}
+	case common.ValueTypeUint32:
+		val, err = cast.ToUint32E(reading)
+		if err != nil {
+			return nil, fmt.Errorf(castError, resource.Name, err)
+		}
+		if err := checkUintValueRange(valueType, val); err != nil {
+			return nil, err
+		}
+	case common.ValueTypeUint64:
+		val, err = cast.ToUint64E(reading)
+		if err != nil {
+			return nil, fmt.Errorf(castError, resource.Name, err)
+		}
+		if err := checkUintValueRange(valueType, val); err != nil {
+			return nil, err
+		}
+	case common.ValueTypeInt8:
+		val, err = cast.ToInt8E(reading)
+		if err != nil {
+			return nil, fmt.Errorf(castError, resource.Name, err)
+		}
+		if err := checkIntValueRange(valueType, val); err != nil {
+			return nil, err
+		}
+	case common.ValueTypeInt16:
+		val, err = cast.ToInt16E(reading)
+		if err != nil {
+			return nil, fmt.Errorf(castError, resource.Name, err)
+		}
+		if err := checkIntValueRange(valueType, val); err != nil {
+			return nil, err
+		}
+	case common.ValueTypeInt32:
+		val, err = cast.ToInt32E(reading)
+		if err != nil {
+			return nil, fmt.Errorf(castError, resource.Name, err)
+		}
+		if err := checkIntValueRange(valueType, val); err != nil {
+			return nil, err
+		}
+	case common.ValueTypeInt64:
+		val, err = cast.ToInt64E(reading)
+		if err != nil {
+			return nil, fmt.Errorf(castError, resource.Name, err)
+		}
+		if err := checkIntValueRange(valueType, val); err != nil {
+			return nil, err
+		}
+	case common.ValueTypeFloat32:
+		val, err = cast.ToFloat32E(reading)
+		if err != nil {
+			return nil, fmt.Errorf(castError, resource.Name, err)
+		}
+		if err := checkFloatValueRange(valueType, val); err != nil {
+			return nil, err
+		}
+	case common.ValueTypeFloat64:
+		val, err = cast.ToFloat64E(reading)
+		if err != nil {
+			return nil, fmt.Errorf(castError, resource.Name, err)
+		}
+		if err := checkFloatValueRange(valueType, val); err != nil {
+			return nil, err
+		}
+	default:
+		return nil, fmt.Errorf("return result fail, unsupported value type: %v", valueType)
+	}
+
+	return val, nil
+}
+
 // HandleReadCommands triggers a protocol Read operation for the specified device.
-func (driver *RestDriver) HandleReadCommands(deviceName string, protocols map[string]models.ProtocolProperties, reqs []dsModels.CommandRequest) ([]*dsModels.CommandValue, error) {
-	return nil, fmt.Errorf("RestDriver.HandleReadCommands; read commands not supported")
+func (driver *RestDriver) HandleReadCommands(deviceName string, protocols map[string]models.ProtocolProperties, reqs []dsModels.CommandRequest) (responses []*dsModels.CommandValue, err error) {
+
+	driver.logger.Debugf("In HandleReadCommands function")
+	var reading interface{}
+	var val interface{}
+	var result = &dsModels.CommandValue{}
+	var uri string
+
+	// Declare response structure,
+	// This will be filled with response received from HTTP GET request
```

Comment:
Please limit comments to `why` something is being done that can not be figured out be reading the code.  i.e. this line doesn't need a comment. Please apply this to all other comments.